src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/DoNotExposeGenericListsTests.cs

```diff
@@ -0,0 +1,274 @@
+            Return Nothing
+        End Function
+    End Structure
+End Namespace
+");
+        }
+
+        [Fact]
+        public async Task CA1002_Properties()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System.Collections.Generic;
+
+namespace Namespace1
+{
+    public interface Interface1
+    {
+        List<int> {|CA1002:InterfaceProperty|} { get; }
+    }
+
+    public class BaseClass
+    {
+        public virtual List<string> {|CA1002:VirtualProperty|} { get; }
+    }
+
+    public class Class1 : BaseClass, Interface1
+    {
+        public List<int> InterfaceProperty { get; } // no issue as this is an inteface implementation
+        public override List<string> VirtualProperty { get; } // no issue this is an override
+
+        public List<int> {|CA1002:AutoProperty|} { get; set; }
+        public List<int> {|CA1002:ReadonlyAutoProperty|} { get; }
+        public List<int> {|CA1002:ReadonlyArrowProperty|} => null;
+
+        private List<int> PrivateAutoProperty { get; set; }
+    }
+
+    public struct Struct1 : Interface1
+    {
+        public List<int> InterfaceProperty { get; }
+
+        public List<int> {|CA1002:ReadonlyArrowProperty|} => null;
+        private List<int> PrivateAutoProperty { get; set; }
+    }
+}");
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Imports System.Collections.Generic
+
+Namespace Namespace1
+    Public Interface Interface1
+        ReadOnly Property {|CA1002:InterfaceProperty|} As List(Of Integer)
+    End Interface
+
+    Public Class BaseClass
+        Public Overridable ReadOnly Property {|CA1002:VirtualProperty|} As List(Of String)
+    End Class
+
+    Public Class Class1
+        Inherits BaseClass
+        Implements Interface1
+
+        Public ReadOnly Property InterfaceProperty As List(Of Integer) Implements Interface1.InterfaceProperty ' no issue as this is an inteface implementation
+        Public Overrides ReadOnly Property VirtualProperty As List(Of String) ' no issue this is an override
+
+        Public Property {|CA1002:AutoProperty|} As List(Of Integer)
+        Public ReadOnly Property {|CA1002:ReadonlyAutoProperty|} As List(Of Integer)
+
+        Private Property PrivateAutoProperty As List(Of Integer)
+    End Class
+
+    Public Structure Struct1
+        Implements Interface1
+
+        Public ReadOnly Property InterfaceProperty As List(Of Integer) Implements Interface1.InterfaceProperty
+
+        Public ReadOnly Property {|CA1002:ReadonlyAutoProperty|} As List(Of Integer)
+
+        Private Property PrivateAutoProperty As List(Of Integer)
+    End Structure
+End Namespace
+");
+        }
+
+        [Fact]
+        public async Task CA1002_ExtensionMethods()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System.Collections.Generic;
+
+namespace Namespace1
+{
+    public static class Helper
+    {
+        public static bool Ext1(this List<int> list) => true;
```

Comment:
This exclusion is not part of the original implementation but that seems weird to get a penalty to extend `List<T>` so I added an exclusion. Please feel free to let me know if you think this shouldn't be done.